src/ui/components/ReplayAssistButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+
+import { setSelectedPrimaryPanel } from "ui/actions/layout";
+import { toggleReplayAssist } from "ui/reducers/app";
+import useAuth0 from "ui/utils/useAuth0";
+
+const ReplayAssistButton: FC = () => {
+  const { isAuthenticated } = useAuth0();
+  const [shouldShowReplayAssist, setShouldShowReplayAssist] = useState(false);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const storedValue = localStorage.getItem("replayAssistEnabled");
+    if (storedValue) {
+      setShouldShowReplayAssist(JSON.parse(storedValue));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem("replayAssistEnabled", JSON.stringify(shouldShowReplayAssist));
+  }, [shouldShowReplayAssist]);
```

Comment:
You can simplify the local storage integration here with the `useLocalStorage` hook:
```suggestion
const dispatch = useDispatch();

const [shouldShowReplayAssist, setShouldShowReplayAssist] =
  useLocalStorage<boolean>(`Replay:replayAssistEnabled`, false);
```

It looks like there's also a syncing bug here though. The Redux state defaults to `false` but the component state defaults to whatever the most recent value in local storage was. These two could get out of sync, which is problematic.